modules/payment-bot/src/index.ts

```diff
@@ -218,16 +167,22 @@ async function run(): Promise<void> {
     const provider = new JsonRpcProvider(config.ethProviderUrl);
     const preWithdrawBal = await provider.getBalance(config.recipient || client.freeBalanceAddress);
     console.log(`Found prewithdrawal balance of ${formatEther(preWithdrawBal)}`);
-    client.on(CFCoreTypes.EventName.WITHDRAWAL_CONFIRMED, async (data: any) => {
-      console.log(`Caught withdraw confirmed event, data: ${JSON.stringify(data, replaceBN, 2)}`);
-      const postWithdrawBal = await provider.getBalance(
-        config.recipient || client.freeBalanceAddress,
-      );
-      console.log(`Found postwithdrawal balance of ${formatEther(postWithdrawBal)}`);
-    });
-    client.on(CFCoreTypes.EventName.WITHDRAWAL_FAILED, async (data: any) => {
-      console.log(`Withdrawal failed with data: ${JSON.stringify(data, replaceBN, 2)}`);
-    });
+    client.on(
+      CFCoreTypes.EventName.WITHDRAWAL_CONFIRMED,
+      async (data: any): Promise<void> => {
+        console.log(`Caught withdraw confirmed event, data: ${JSON.stringify(data, replaceBN, 2)}`);
+        const postWithdrawBal = await provider.getBalance(
+          config.recipient || client.freeBalanceAddress,
+        );
+        console.log(`Found postwithdrawal balance of ${formatEther(postWithdrawBal)}`);
+      },
+    );
+    client.on(
+      CFCoreTypes.EventName.WITHDRAWAL_FAILED,
+      async (data: any): Promise<void> => {
+        console.log(`Withdrawal failed with data: ${JSON.stringify(data, replaceBN, 2)}`);
+      },
+    );
```

Comment:
we should hard error if any of these `failed` events come through, since as it stands bots will continue to execute and CI won't catch it